lib/charms/kubernetes_charm_libraries/v0/kubernetes_hugepages_volumes_patch.py

```diff
@@ -0,0 +1,626 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm library used to manage HugePages volumes in Kubernetes charms.
+
+- On bound event (e.g. self.on.hugepages_volumes_config_changed which is originated
+from K8sHugePagesVolumePatchChangedEvent), it will:
+  - Replace the volumes in the StatefulSet with the new requested ones
+  - Replace the volume mounts in the container in the Pod with the new requested ones.
+  - Replace the resource requirements in the container in the Pod with the new requested ones.
+
+## Usage
+
+```python
+
+from charms.kubernetes_charm_libraries.v0.kubernetes_hugepages_volumes_patch import (
+    KubernetesHugePagesPatchCharmLib,
+    RequestedHugePages,
+)
+
+
+class K8sHugePagesVolumePatchChangedEvent(EventBase):
+    def __init__(self, handle: Handle):
+        super().__init__(handle)
+
+
+class K8sHugePagesVolumePatchChangedCharmEvents(CharmEvents):
+    hugepages_volumes_config_changed = EventSource(K8sHugePagesVolumePatchChangedEvent)
+
+
+class YourCharm(CharmBase):
+
+    on = K8sHugePagesVolumePatchChangedCharmEvents()
```

Comment:
I know this is not specific to this charm, but what if there is more than one `on` that we need, like say for both hugepages and multus?

Reply:
We agreed to create a new class `UpfOperatorCharmEvents(CharmEvents)` which provides attributes for both multus events (`nad_config_changed`) and (`hugepages_volumes_config_changed`).